File: src/test/kotlin/org/taktik/icure/entities/PropertyTest.java
package org.taktik.icure.entities;

import org.junit.Test;
import org.taktik.icure.services.external.rest.handlers.GsonMessageBodyHandler;
import org.taktik.icure.services.external.rest.v1.dto.PropertyDto;
import org.taktik.icure.services.external.rest.v1.transformationhandlers.V1MapperFactory;

import static org.junit.Assert.*;

public class PropertyTest {
    @Test
    public void map() {
        PropertyDto result = new V1MapperFactory(new GsonMessageBodyHandler().getGson()).getMapper().map(new Property(new PropertyType(), 0), PropertyDto.class);
        assertNotNull(result);
    }
}

File: src/test/kotlin/org/taktik/icure/entities/PropertyTest.kt
package org.taktik.icure.entities

import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import org.mapstruct.factory.Mappers
import org.taktik.icure.entities.embed.TypedValue
import org.taktik.icure.services.external.rest.v1.dto.PropertyDto
import org.taktik.icure.services.external.rest.v1.mapper.PropertyMapper

class PropertyTest {
    @Test
    fun map() {
        val result: PropertyDto = Mappers.getMapper(PropertyMapper::class.java).map(Property(id= "123", type = PropertyType(id = "123"), typedValue = TypedValue.withValue(0)))
        assertNotNull(result)
    }
}
